controllers/helmchartinstall_controller.go

```diff
@@ -77,61 +126,255 @@ func (r *HelmChartInstallReconciler) render(ctx context.Context, claim *helmv1al
+	configuration, err := r.getStackConfiguration(ctx, claim)
+	// TODO check for errors
+
+	targetResourceBehavior, err := r.getBehavior(ctx, claim, configuration)
+
+	if targetResourceBehavior == nil {
+		// TODO error condition with a real error returned
+		r.Log.V(0).Info("Couldn't find a configured behavior!",
+			"claim", claim,
+			"configuration", configuration,
+		)
+		return err
+	}
+
+	engineConfig, err := r.createBehaviorEngineConfiguration(ctx, claim, configuration)
+
+	if err != nil {
+		r.Log.Error(err, "Error creating engine configuration!", "claim", claim)
+		return err
+	}
+
+	targetStackImage := configuration.Spec.Source.Image
 
-	// TODO the status will have more than *just* the result in it,
-	// but we can start with just the result
-	result, err := r.processConfiguration(ctx, claim, configuration)
+	result, err := r.executeBehavior(ctx, claim, engineConfig, targetStackImage, targetResourceBehavior)
+	// TODO check for errors
 
 	err = r.setClaimStatus(claim, result)
 
 	return err
 }
 
-func (r *HelmChartInstallReconciler) getClaim() (unstructured.Unstructured, error) {
-	// The claim is the CR that triggered this whole thing.
-	return unstructured.Unstructured{}, nil
-}
-
-func (r *HelmChartInstallReconciler) getStackConfiguration() (unstructured.Unstructured, error) {
+func (r *HelmChartInstallReconciler) getStackConfiguration(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+) (*v1alpha1.StackConfiguration, error) {
 	// See the template stacks internal design doc for details, but
 	// the most likely source of the stack configuration is the stack object itself.
 	// Other potential sources include a configmap
 
 	// TODO
-	// - Stack configuration should be a structured object
 	// - Stack configuration will be coming from a Kubernetes object, probably the
 	//   Stack itself
-	//stackConfiguration := ''
-	return unstructured.Unstructured{}, nil
+	namespacedName, err := client.ObjectKeyFromObject(claim)
+
+	if err != nil {
+		r.Log.V(0).Info("getStackConfiguration returning early because of error creating object key", "err", err, "claim", claim)
+		return nil, err
+	}
+
+	configuration := &v1alpha1.StackConfiguration{}
+	if err := r.Client.Get(ctx, namespacedName, configuration); err != nil {
+		r.Log.V(0).Info("getStackConfiguration returning early because of error fetching configuration", "err", err, "claim", claim)
+		return nil, err
+	}
+
+	r.Log.V(0).Info("getStackConfiguration returning configuration", "configuration", configuration)
+	return configuration, nil
+}
+
+/**
+ * When a behavior is triggered, we want to know which behavior exactly we are executing.
+ *
+ * In most cases, this will probably be configured ahead of time by the setup controller, rather
+ * than being fetched at runtime by the render controller.
+ */
+func (r *HelmChartInstallReconciler) getBehavior(
+	ctx context.Context,
+	claim *unstructured.Unstructured,
+	configuration *v1alpha1.StackConfiguration,
```

Comment:
Per the whole "short variable names in Go" thing I would name `configuration` `c` or `cfg` in this method. I can see from its type that it is a `StackConfiguration`, and this method is not so long that I ever really lose that context as a reader.

Conversely, `claim` is a great name for the `*unstructured.Unstructured`, because its type gives me no context on what it it means to this method.